src/lib.rs

```diff
@@ -284,11 +282,11 @@ impl Ngram {
     /// Private method that initializes an `Ngram` by calculating all of its
     /// grams.
     fn init(&mut self) {
-        if self.arity > self.text_padded.len() {
+        if ARITY > self.text_padded.len() {
```

Comment:
ARITY of 0 is invalid - we should put an assert or panic in here if ARITY == 0.

Reply:
Maybe I can switch from an ARITY const generic to struct markers - as I understand generally speaking ARITY will be in a rather limited set (1, 2, 3, 4 - hardly larger) and avoid the need for a check.

Reply:
Hmmm... this might be premature optimization, but another way we could get the same result but without doing the check at runtime for a compile-time value would be to

```
impl Ngram<0> {
    fn init(...) {
        // Ideally compile-time assert or panic, but runtime is an acceptable fallback
    }
}

impl NgramBuilder<0> {
   pub fn new(...) {
        // Ideally compile-time assert or panic, but runtime is an acceptable fallback
    }
}
```